x/harvest/keeper/borrow.go

```diff
@@ -19,11 +19,31 @@ func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, coins sdk.Coins
 		}
 	}
 
-	// Sync user's borrow balance (only for coins user is requesting to borrow)
-	k.SyncBorrowInterest(ctx, borrower, coins)
+	// Get current stored LTV based on stored borrows/deposits
+	prevLtv, err := k.GetCurrentLTV(ctx, borrower)
+	if err != nil {
+		return err
+	}
+
+	// We use a list of BorrowIndexItem here because Amino doesn't support marshaling maps.
+	_, found := k.GetBorrow(ctx, borrower)
+	if !found {
+		// On user's first borrow, build borrow index list containing denoms and current global borrow index value
+		var borrowIndexes types.BorrowIndexes
+		for _, coin := range coins {
+			borrowIndexValue, _ := k.GetBorrowIndex(ctx, coin.Denom)
+			borrowIndex := types.NewBorrowIndexItem(coin.Denom, borrowIndexValue)
+			borrowIndexes = append(borrowIndexes, borrowIndex)
+		}
+		borrow := types.NewBorrow(borrower, sdk.Coins{}, borrowIndexes)
+		k.SetBorrow(ctx, borrow)
```

Comment:
I believe `SetBorrow` is only safe to call after `SendCoinsFromModuleToAccount` or some equivalent check like a `validateBalance` function is called and `validateBorrow` has passed. As currently written, the borrow would get created in the store even if the user didn't have `coins` in their account or the LTV was invalid.

Reply:
Fixed in https://github.com/Kava-Labs/kava/pull/742/commits/e32ec93b237a7a0f5a4f36110c5d72a6162f9cfe